### PR TITLE
Learn New Words

### PR DESCRIPTION
**Pull request by CharlieHess**
This PR exposes an `add` API for all platforms, so that users can improve their dictionary over time. Also went ahead and updated some documentation.

- [x] Verify on Mac
- [x] Verify on Windows (using Hunspell)

### CODE REVIEW

**Comment by paulcbetts** [comment]
Do these adds persist for >1 session on all platforms? Where does Hunspell store the custom dictionary?

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "main": "./lib/spellchecker.js",
   "name": "spellchecker",
   "description": "Bindings to native spellchecker",
-  "version": "2.1.0",
+  "version": "2.2.0",
```
Comment (kevinsawicki) [inline package.json]: This will be bumped automatically when it is published, please undo this change.

File: src/main.cc [hunk 2]
```diff
@@ -49,6 +49,19 @@ class Spellchecker : public ObjectWrap {
     NanReturnValue(NanNew(that->impl->IsMisspelled(word)));
   }
 
+  static NAN_METHOD(Add) {
+    NanScope();
+    if (args.Length() < 1) {
+      return NanThrowError("Bad argument");
```
Comment (kevinsawicki) [inline src/main.cc]: Mind adding a spec for this case as well?

Comment (CharlieHess) [inline src/main.cc]: Totally. :+1:

**Comment by CharlieHess** [comment]
@paulcbetts: Using Mac and Windows 8, you're all good. But when using Hunspell, the adds do not persist. Currently this is an Exercise For The User- Hunspell doesn't do anything on its own. Do you think we should manage the custom dictionary in this module?

**Comment by paulcbetts** [comment]
@CharlieHess We should save it somewhere - I bet Hunspell lets you specify a user dictionary path that they'll maintain

**Comment by CharlieHess** [comment]
@paulcbetts: Not AFAICT. We can create a new dictionary file and pass that to Hunspell, but it's on us to maintain it.

**Comment by kevinsawicki** [comment]
Thanks for this

**Comment by kevinsawicki** [comment]
Published in 2.2.0
